rust/src/utils/gas.rs

```diff
@@ -27,35 +27,202 @@ pub fn consume_positive_value_cost(value: &u256, gas_left: &mut u64) -> Result<(
 pub fn consume_value_to_empty_account_cost<E: ExecutionContextTrait>(
     value: &u256,
     addr: &Address,
-    state: &mut Interpreter<E>,
+    interpreter: &mut Interpreter<E>,
 ) -> Result<(), StatusCode> {
-    if *value != u256::ZERO && !state.context.account_exists(addr) {
-        consume_gas(&mut state.gas_left, 25000)?;
+    if *value != u256::ZERO && !interpreter.context.account_exists(addr) {
+        consume_gas(25_000, &mut interpreter.gas_left)?;
     }
     Ok(())
 }
 
 #[inline(always)]
 pub fn consume_address_access_cost<E: ExecutionContextTrait>(
     addr: &Address,
-    state: &mut Interpreter<E>,
+    interpreter: &mut Interpreter<E>,
 ) -> Result<(), StatusCode> {
-    if state.revision < Revision::EVMC_BERLIN {
+    if interpreter.revision < Revision::EVMC_BERLIN {
         return Ok(());
     }
-    if state.context.access_account(addr) == AccessStatus::EVMC_ACCESS_COLD {
-        consume_gas(&mut state.gas_left, 2600)
+    if interpreter.context.access_account(addr) == AccessStatus::EVMC_ACCESS_COLD {
+        consume_gas(2_600, &mut interpreter.gas_left)
     } else {
-        consume_gas(&mut state.gas_left, 100)
+        consume_gas(100, &mut interpreter.gas_left)
     }
 }
 
-/// consume 3 * minimum_word_size
 #[inline(always)]
-pub fn consume_copy_cost(gas_left: &mut u64, len: u64) -> Result<(), StatusCode> {
-    let (cost, cost_overflow) = word_size(len)?.overflowing_mul(3);
-    if cost_overflow {
-        return Err(StatusCode::EVMC_OUT_OF_GAS);
+pub fn consume_copy_cost(len: u64, gas_left: &mut u64) -> Result<(), StatusCode> {
```

Comment:
I am not sure if we need a function at all. It would not be too large if you write:
` consume_gas(word_size(len)? * 3, &mut gas)`,

Reply:
I find it better this way because it is self documenting.